### PR TITLE
Switch to System.Text.Json

### PR DESCRIPTION
**Pull request by danielmarbach**
Alternative to #152 

Which would be my preference. Given that we can release it as 0.4 and then slowly adjust approval files when necessary, I think we should switch the serializer

### CODE REVIEW

File: src/Particular.Approvals/Particular.Approvals.csproj [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-    <TargetFrameworks>net452;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net462;netstandard2.0</TargetFrameworks>
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>$(SolutionDir)NServiceBus.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Newtonsoft.Json" Version="[13.0.1, 14.0.0)" />
+    <PackageReference Include="System.Text.Json" Version="7.0.0" />
     <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/Particular.Approvals/Particular.Approvals.csproj:11]: We should go ahead and update this to latest as well, though I guess that could be done in a separate PR.

File: src/Tests/Tests.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net452;netcoreapp3.1;net6.0</TargetFrameworks>
+    <TargetFrameworks>net462;net6.0;net7.0</TargetFrameworks>
```
Comment (bording) [inline src/Tests/Tests.csproj:4]:
```suggestion
    <TargetFrameworks>net472;net6.0;net7.0</TargetFrameworks>
```

File: src/Particular.Approvals/Particular.Approvals.csproj [hunk 3]
```diff
@@ -1,13 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net452;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net462;netstandard2.0</TargetFrameworks>
```
Comment (bording) [inline src/Particular.Approvals/Particular.Approvals.csproj:4]: If we're updating this and not worrying about older release branches, we might as well bump this up to `net472`.

```suggestion
    <TargetFrameworks>net472;netstandard2.0</TargetFrameworks>
```

**Review by bording — CHANGES_REQUESTED** [review]

**Comment by danielmarbach** [comment]
@bording feel free to make those changes by directly pushing here. I'm 👍

**Review by bording — APPROVED** [review]

**Pull request merged**